packages/data-provider/src/utils/fileUtils.ts

```diff
@@ -90,6 +165,19 @@ export function getSizeInBytes(size: string | number): number {
 	return bytes.parse(size);
 }
 
+/**
+ * Formats a file size from bytes to a specified unit with a defined precision.
+ *
+ * @param sizeInBytes - The file size in bytes to be formatted.
+ * @param unit - The unit to which the size should be converted (e.g., 'MB', 'GB').
+ * @param precision - The number of decimal places to include in the formatted output.
+ * @returns The file size formatted as a string in the specified unit with the given precision.
+ *
+ */
+export const formatByteSize = (sizeInBytes: number, unit: Unit, precision: number) => {
```

Comment:
Great description, very helpful! Thanks for using types to restrict the unit values!